### DISCUSSION TITLE
Screen rotation lock

### DISCUSSION THREAD

**Discussion by freimenue** [seerge/g-helper#2071] [Ideas]
Hi all, 

disclaimer: only created a github acc to ask this, not customary with how things are run around here ;)

I use g-helper on my 2023 Flow X16 (GV601VI) and it has solved a lot of the gripes I had with the device. Big thanks for that. But there is one issue left that regularly annoys me, so I was wondering if it's possible to integrate this into g-helper as well: 

The screen rotation (lock). In the windows quick toggles I can toggle the screen rotation lock only when I rotate the screen more then 180°, so in controller/tent mode and tablet mode. The chosen setting then stays even when I go back to laptop-mode, but cannot be changed until flippping the screen back over. Also rotating the device with rotation lock enabled triggers the device to enable tablet mode (windows taskbar resizes), which is not always ideal. 

Is there a possibility to add toggles to g-helper, that allow me to use screen rotation lock when in laptop mode? 

Many thanks!

**Comment by seerge** [comment]
@freimenue hello, that's something windows handles by itself (mb by getting notifications from firmware about actual rotation)

I don't think that AC could control that neither, so I can't really add that in G-Helper :)

**Reply by freimenue** [reply]
Aah, that's a shame. But thanks anyway for the quick reply!

**Comment by NyaMisty** [comment]
I've got Flow X16 and is facing exactly the same issue. I'm looking into windows internals to see why it got into tablet mode all the times....

**Comment by NyaMisty** [comment]
After some digging, I found the following:
1. rotation lock can only be used in tablet mode, and it's useless on laptop mode.
2. you can disable the automatic tablet switch by disabling the corresponding mode-switching driver (see https://learn.microsoft.com/en-us/windows/client-management/client-tools/manage-device-installation-with-group-policy), and disabling "Device Instance ID" with "ACPI\INT33D3\2&DABA3FF&0" and "Device Id" with "*INT33D3"
3. after disabling that, if you still want to switch tablet mode, you can either re-enable the switch by disabling the above policy, or you can manually change the `ConvertibleSlateMode` to 0 under `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\PriorityControl`

With the above tweak, Flow X16 won't enter tablet automatically.

**Reply by NyaMisty** [reply]
However, when you flip the laptop into tent mode, the keyboard will still being disabled.
